cpp/src/parquet/types.h

```diff
@@ -17,6 +17,16 @@
 
 #pragma once
 
+#ifdef _WIN32
+
+// parquet.thrift's OPTIONAL RepetitionType conflicts with a #define from
+// above, so we undefine it
+#ifdef OPTIONAL
+#undef OPTIONAL
+#endif
```

Comment:
Can you move this after the includes? Also, can you update the comment (there is no "define from above")?

Reply:
Done in 1a77246